libbeat/common/transport/tls.go

```diff
@@ -73,6 +73,63 @@ func TestTLSDialer(
 	}), nil
 }
 
+type DialerH2 interface {
```

Comment:
not an issue of this PR but it's pretty disturbing we don't provide Context Dialer

Reply:
I agree, I was trying to limit the change as much as possible. But we should look at switching all of it to context dailers.